mapreduce/src/main/java/org/pentaho/hadoop/mapreduce/OutputCollectorRowListener.java

```diff
@@ -195,7 +202,8 @@ public void rowWrittenEvent( RowMetaInterface rowMeta, Object[] row ) throws Ket
    */
   public void setDebugStatus( Reporter reporter, String message ) {
     if ( debug ) {
```

Comment:
This is not how it should be. This should be:

```
if (log.isDebugEnabled()) {
  log.logDebug( message );
}
```

Do not use "if (debug)" because this is hardcoded and will never be enabled. 

Do not use "setLogLevel()" because you're overriding the system administrator's configuration.

Reply:
I can't not use if (debug) here as debug is an instance variable and used in the constructor:
public OutputCollectorRowListener( OutputCollector<K, V> output, Class<K> outClassK, Class<V> outClassV,
                                     Reporter reporter, boolean debug ) {

and all logic later on depends on debug variable